prime-angular-frontend/src/app/modules/adjudication/shared/components/adjudication-document-overview/adjudication-document-overview.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+
+import { AbstractOverview } from '@lib/classes/abstract-overview.class';
+import { BaseDocument } from '@shared/components/document-upload/document-upload/document-upload.component';
+import { PaperEnrolmentRoutes } from '@paper-enrolment/paper-enrolment.routes';
+import { PaperEnrolmentAgreementTypeNameMap } from '@shared/enums/agreement-type.enum';
+
+@Component({
+  selector: 'app-adjudication-document-overview',
+  templateUrl: './adjudication-document-overview.component.html',
+  styles: [
+    '.mat-icon { font-size: 1.2em; }',
+    '.button > .mat-icon { font-size: 1.35rem; }'
+  ],
+})
+export class AdjudicationDocumentOverviewComponent extends AbstractOverview {
+  @Input() public documents: BaseDocument[];
+  @Output() public download: EventEmitter<{ documentId: number }>;
+  public PaperEnrolmentAgreementTypeNameMap = PaperEnrolmentAgreementTypeNameMap;
+  public PaperEnrolmentRoutes = PaperEnrolmentRoutes;
+  public TOADocuments: BaseDocument[];
+  public supportingDocuments: BaseDocument[];
+  public paperForms: BaseDocument[];
+
+  constructor(
+    route: ActivatedRoute,
+    router: Router
+  ) {
+    super(route, router, PaperEnrolmentRoutes.MODULE_PATH);
+
+    this.download = new EventEmitter<{ documentId: number }>();
+    this.TOADocuments = [];
+    this.supportingDocuments = [];
+    this.paperForms = [];
+  }
+
+  ngOnInit() {
```

Comment:
got `eslint: warning Lifecycle interface 'OnInit' should be implemented for method 'ngOnInit'.`

seems it should be 
```typescript 
public ngOnInit(): void
``` 
and
```typescript
export class AdjudicationDocumentOverviewComponent extends AbstractOverview implements OnInit {
```

Reply:
I did generate it through CLI, I did delete a some of the boilerplate code though so that's may be why I missed it. Fixed